x/distribution/spec/04_messages.md

```diff
@@ -27,163 +30,91 @@ func (k Keeper) SetWithdrawAddr(ctx sdk.Context, delegatorAddr sdk.AccAddress, w
-from the passive global pool to their own pool. It is at this point that the
-commission is withdrawn
-
-```go
-func (vi ValidatorDistInfo) TakeFeePoolRewards(g FeePool, height int64, totalBonded, vdTokens, commissionRate Dec) (
-                                vi ValidatorDistInfo, g FeePool)
-
-    g.UpdateTotalValAccum(height, totalBondedShares)
-    
-    // update the validators pool
-    blocks = height - vi.FeePoolWithdrawalHeight
-    vi.FeePoolWithdrawalHeight = height
-    accum = blocks * vdTokens
-    withdrawalTokens := g.Pool * accum / g.TotalValAccum 
-    commission := withdrawalTokens * commissionRate
-    
-    g.TotalValAccum -= accumm
-    vi.PoolCommission += commission
-    vi.PoolCommissionFree += withdrawalTokens - commission
-    g.Pool -= withdrawalTokens
-
-    return vi, g
-```
+The validator can send the WithdrawValidatorCommission message to withdraw their accumulated commission.
+The commission is calculated in every block during `BeginBlock`, so no iteration is required to withdraw.
+The amount withdrawn is deducted from the `ValidatorOutstandingRewards` variable for the validator.
+Only integer amounts can be sent. If the accumulated awards have decimals, the amount is truncated before the withdrawal is sent, and the remainder is left to be withdrawn later.
 
+## FundCommunityPool
 
-### Delegation reward withdrawal
+This message sends coins directly from the sender to the community pool.
 
-For delegations (including validator's self-delegation) all rewards from reward
-pool have already had the validator's commission taken away.
+The transaction fails if the amount cannot be transferred from the sender to the distribution module account.
 
 ```go
-func (di DelegationDistInfo) WithdrawRewards(g FeePool, vi ValidatorDistInfo,
-    height int64, totalBonded, vdTokens, totalDelShares, commissionRate Dec) (
-    di DelegationDistInfo, g FeePool, withdrawn DecCoins)
-
-    vi.UpdateTotalDelAccum(height, totalDelShares) 
-    g = vi.TakeFeePoolRewards(g, height, totalBonded, vdTokens, commissionRate) 
-    
-    blocks = height - di.WithdrawalHeight
-    di.WithdrawalHeight = height
-    accum = delegatorShares * blocks 
-     
-    withdrawalTokens := vi.Pool * accum / vi.TotalDelAccum
-    vi.TotalDelAccum -= accum
-
-    vi.Pool -= withdrawalTokens
-    vi.TotalDelAccum -= accum
-    return di, g, withdrawalTokens
+func (k Keeper) FundCommunityPool(ctx sdk.Context, amount sdk.Coins, sender sdk.AccAddress) error {
+    if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, sender, types.ModuleName, amount); err != nil {
+        return err
+    }
 
-```
+	feePool := k.GetFeePool(ctx)
+	feePool.CommunityPool = feePool.CommunityPool.Add(sdk.NewDecCoinsFromCoins(amount...)...)
+	k.SetFeePool(ctx, feePool)
 
-### Validator commission withdrawal
+	return nil
+}
+```
 
-Commission is calculated each time rewards enter into the validator.
+## Common distribution operations
```

Comment:
instead of stacking headings, can you introduce what common distribution operations are?

Reply:
```suggestion
## Common distribution operations

These operations take place during many different messages.
```